### PR TITLE
feat(discover): Add avg() aggregation function

### PR DESCRIPTION
**Pull request by lynnagara**
Closes APP-396

Depends on https://github.com/getsentry/sentry/pull/8900

### CODE REVIEW

File: src/sentry/static/sentry/app/views/organizationDiscover/aggregations/aggregation.jsx [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+      selectedFunction: null,
+    };
+  }
+
+  getOptions() {
+    const currentValue = getInternal(this.props.value);
+    return [{label: currentValue, value: currentValue}];
+  }
+
+  filterOptions = (options, input, value) => {
```
Comment (billyvg) [inline src/sentry/static/sentry/app/views/organizationDiscover/aggregations/aggregation.jsx]: Can we have add tests for this?

File: src/sentry/static/sentry/app/views/organizationDiscover/aggregations/aggregation.jsx [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+    }
+
+    return optionList.filter(({label}) => label.includes(input));
+  };
+
+  focus() {
+    this.select.focus();
+  }
+
+  handleChange = option => {
```
Comment (billyvg) [inline src/sentry/static/sentry/app/views/organizationDiscover/aggregations/aggregation.jsx]: Let's add assertions for component state for the different branches below, and also a test for when `onChange` gets called.

File: src/sentry/static/sentry/app/views/organizationDiscover/aggregations/aggregation.jsx [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+
+  inputRenderer = props => {
+    const val = `${this.state.selectedFunction || ''}`.trim();
+
+    return (
+      <input
+        type="text"
+        {...props}
+        value={props.value || val}
+        style={{width: '100%', border: 0}}
```
Comment (billyvg) [inline src/sentry/static/sentry/app/views/organizationDiscover/aggregations/aggregation.jsx]: Can you make this a styled component?

Comment (lynnagara) [reply src/sentry/static/sentry/app/views/organizationDiscover/aggregations/aggregation.jsx]: it's not super straightforward with react select, i'd come back and look closer at it later when tackling styling

**Review by billyvg — APPROVED** [review]

**Pull request merged**